Tournament.cpp

```diff
@@ -0,0 +1,263 @@
+#include "Tournament.h"
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+Tournament::Tournament(int choice, std::string& path, std::string& pathGroup)
+{
+  const int EURO = 1;
+  const int WORLD = 2;
+  if (choice == EURO)
+  {
+	try
+	{
+	  addParticipants(path, m_euroParticipants);
+	  setGroup(pathGroup, m_numberEuroGroup);
```

Comment:
It appears to me that making two subclasses of Tournament, EuroTournament and WorldTournament, and putting corresponding logic to them might be a good choice.